content/protocols/mqtt.textile

```diff
@@ -0,0 +1,238 @@
+
+Currently both "rewind":/channels/options/rewind and "delta":/channels/options/delta channel options are supported.
+
+h3(#delta-mqtt). Delta with MQTT
+
+If subscribing to a channel in delta mode using MQTT then you will need to decode any received delta messages yourself.
+
+When subscribing without an Ably library, the channel @delta@ parameter must be specified using a qualified channel name. The qualifier part is in square brackets at the start of the channel name.
+
+To specify the parameter @foo@ with value @bar@ on channel @baz@, the qualified channel name would be @[?foo=bar]baz@. If the channel name already has a qualifier, such as @[meta]log@, then the query string follows the existing qualifier, as in @[meta?foo=bar]log@.
+
+Some transports provide raw message payloads - that is, the content of the @data@ attribute of a @Message@ - without the accompanying metadata. That means that the recipient of the message does not have access to the @extras@ or @encoding@ attributes of the message that would ordinarily be used to decode delta message payloads.
+
+In order to assist applications that use these transports, @vcdiff@ decoder libraries can check for the @vcdiff@ header at the start of the message payload as an inexact method of determining whether or not the message is a regular message or a delta. Note that, in order to rely on that check, you need to know that that header will not be present in any valid (uncompressed) message in your app. No valid JSON value, for example, will match the @vcdiff@ header check, so it is safe to perform this sniffing on JSON message payloads.
+
+Read more in the "delta section":/channels/options/delta.
+
+h4(#delta-example). Delta example
+
+```[nodejs]
+  var mqtt = require('mqtt');
+  var { VcdiffDecoder } = require('@ably/vcdiff-decoder');
+
+  var options = {
+    keepalive: 30,
+    username: '{{API_KEY_NAME}}', /* API key's name */
+    password: '{{API_KEY_SECRET}}', /* API key's secret */
+    port: 8883
+  };
+  var client = mqtt.connect('mqtts:mqtt.ably.io', options);
+  var channelName = 'sample-app-mqtt';
+  var channelDecoder = new VcdiffDecoder();
+
+  client.on('message', (_, payload) => {
+    var data = payload;
+
+    try {
+      if (VcdiffDecoder.isDelta(data)) {
+        data = channelDecoder.applyDelta(data).asUint8Array();
+      } else {
+        channelDecoder.setBase(data);
+      }
+    } catch(e) {
+      /* Delta decoder error */
+      console.log(e);
+    }
+
+    /* Process decoded data */
+    console.log(data);
+  });
+
+  client.subscribe(`[?delta=vcdiff]${channelName}`);
+```
+
+h3(#rewind-mqtt). Rewind with MQTT
+
+To use rewind with MQTT you specify the @rewind@ channel option with a numerical value equal to the position you want to attach to, for example @"1"@ to rewind 1 message previous. This will apply to all channels specified in the query string.
```

Comment:
```suggestion
The @rewind@ channel option enables a client to specify where to start an attachment from. This can be a point in time in the past, or a given number of messages.
```